src/core/CompUnit/Repository/Installation.pm6

```diff
@@ -430,10 +441,10 @@ sub MAIN(:$name, :$auth, :$ver, *@, *%) {
         }
 
         # Sort from highest to lowest by version and api
-        my $sorted-metas := $matching-metas.sort(*.value<api>).sort(*.value<ver>).reverse;
+        my $sorted-metas := $matching-metas.sort({ .value<name> eq $spec.short-name }).sort(*.value<api>).sort(*.value<ver>).reverse;
```

Comment:
Won't this give warnings about uninitialized `.value<name>`?

Reply:
I suppose for upgraded (instead of freshly installed) rakudo yes. `name` is a new field added to the short lookup:
https://github.com/rakudo/rakudo/blob/02c46a96b403c73d9fef68f99db5fd8a1bfde6ab/src/core/CompUnit/Repository/Installation.pm6#L424

I'll need to add some logic to `upgrade-repository` to add the extra field in the short-name-lookup file. Since https://github.com/rakudo/rakudo/pull/2778 also need to upgrade the repository version it would probably be wise for these two PRs to be rolled into one once the details are ironed out.